tests/attributeChangedCallback.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { expect } from 'chai';
+import Component from '../src/index';
+
+describe('attributeChangedCallback', () => {
+    it('does not let an unwatched attribute pass through to props', () => {
+        class Test extends Component<{ myAttribute: string }, {}> {
+            static componentName = 'x-test';
+        }
+        const testComponent = Object.create(Test.prototype, {});
```

Comment:
A: This would require us to define every class existence in the spec files. When using the new keyword, we get an Illegal Constructor error, because the constructor is not known to typescript.